### ISSUE TITLE
Consider framework and architecture in updates

### ISSUE THREAD

**Issue by mardy** [UbuntuOpenStore/openstore-meta#161]
If my understanding of the `/api/v2/apps/revision/` endpoint is correct, the openstore is currently assuming that applications will never change their base framework and architecture.

Frameworks can change easily as the platform evolves, but also architectures can be added or removed to clicks as new devices appear (or the developer decides to support/discontinue some architecture). I would suggest that the api should accept also `architecture` and `frameworks` parameter.

**Comment by bhdouglass** [comment]
@sverzegnassi how does the app currently handle these situations?

**Comment by sverzegnassi** [comment]
Client-app always attaches information about architecture and frameworks to any request sent to the server, even if the parameters are not supported by the server yet.

However, the client does not perform any further check, it just takes the reply for the server and compare local and remote revisions. That means that, in mardy's example, the client may return false positives.

- Architecture. Currently I haven't implemented any detection yet, that was missing even before I started to rewrite the backend. `armhf` value is hardcoded into the sources, but clearly needs fixing.
- Frameworks. Filtering by framework is supported by the client-app only where the server accepts 'frameworks' as parameter. I suspect that  `discover`, `categories`, `revision`, and `updates` endpoints still don't support the parameter.

**Comment by bhdouglass** [comment]
Ok, sounds like the discover/categories/revision api endpoint need to be updated to account for arch & framework. And we can get rid of the updates api (since that was the precursor to the revision endpoint anyways).

**Comment by mardy** [comment]
A separate but related request would be to return the full package manifest in the reply, and not just version/revision info. At least we need title, description and icon because they might have been changed, size and changelog. I don't think we need more than this, but it might be simpler to return the whole object anyway.
Would you approve such a change, if I submit it?

**Comment by bhdouglass** [comment]
@mardy yeah, but it might make sense to have it be a separate endpoint (to keep the current one that the app uses light).

**Comment by mardy** [comment]
@bhdouglass maybe we can add a GET parameter to decide what data to retrieve? For example:
 * `fields=none`: current (and default) behaviour
 * `fields=all`: return full manifest

This leaves us the opportunity to later implement a way for the client to list all the fields it's interested in, should the need arise.

**Comment by bhdouglass** [comment]
@mardy yeah that should be fine, I would just want it to be either no `fields=` or `fields=all`

**Comment by JassMan23** [comment]
IMHO OpenStore should display the framework from manifest below the current version for the displayed app. A number of users have had problems after reverting from 16.04rc back to 15.04stable. Displaying framework shows which apps need to be manually downloaded from open-store.io . By default users have no means to view the contents of the manifest file for each individual package, so openStore is the obvious app to show this info.

**Comment by bhdouglass** [comment]
@JassMan23 I'm not sure that will always be helpful. For example may of my apps have the same framework regaurdless of vivid/xenial.
